synapse/handlers/sync.py

```diff
@@ -1817,11 +1817,30 @@ async def _generate_sync_entry_for_rooms(
             )
             sync_result_builder.now_token = now_token
 
+        # Retrieve rooms that got un partial stated in the meantime, only useful in case
+        # of a non lazy-loading-members sync.
+        un_partial_stated_rooms = set()
+        if not sync_result_builder.sync_config.filter_collection.lazy_load_members():
+            un_partial_stated_rooms_since = 0
```

Comment:
Hum after thinking a bit more I think this is dangerous performance wise: on an initial sync we would fetch all the rooms that has ever been un-partial-stated. Not cool.

I am thinking about 2 complementary fixes:
1) limit the store SQL request with the existing `joined_room_ids`
2) do not compute the set at all if we have no `since` token

@DMRobertson I am explicitly tagging you in case of :)

Reply:
Done !

Reply:
> I think this is dangerous performance wise: on an initial sync we would fetch all the rooms that has ever been un-partial-stated. Not cool.

Initial syncs are slow anyway (for exactly this kind of reason, I'd guess: having to scan through all of history?)

Reply:
I'll take a look today

Reply:
Hmm. What do we do for other streams here?

Reply:
> limit the store SQL request with the existing joined_room_ids

I was a little sceptical of this at first, but I see that `matrix.un_partial_stated_room_stream` has an index on `room_id`, so I think this is ok.